### PR TITLE
feat(UItest): Public tests for GHA

### PR DESCRIPTION
**Pull request by martinszuc**
## What is the purpose of this change? What does it change?

Refactored Integration-UI-Test's part of project.

I suggest splitting UI end-to-end tests into public and cluster workflows. I also wanted to outline basic structure for simpler addition of tests in the future.

### CODE REVIEW

File: src/it/java/org/jboss/tools/intellij/openshift/test/ui/AbstractBaseTest.java [hunk 1]
```diff
@@ -32,20 +34,26 @@
...
+    private static boolean hasConnectedToTestIDE = false;
+    public static final String DEFAULT_CLUSTER_URL = "no (current) context/cluster set";
 
     @BeforeAll
     public static void connect() {
-        robot = IdeaRunner.getInstance().getRemoteRobot();
-        ProjectUtility.createEmptyProject(robot, "test-project");
+        if (!hasConnectedToTestIDE) {
+            robot = IdeaRunner.getInstance().getRemoteRobot();
+            hasConnectedToTestIDE = true;
```
Comment (lgrossma) [inline src/it/java/org/jboss/tools/intellij/openshift/test/ui/AbstractBaseTest.java:44]: What is the reason for setting hasConnectedToTestIDE to false and checking its value with condition? The condition always evaluate as True, so it seems a little bit unnecessary.

**Review by lgrossma — CHANGES_REQUESTED** [review]

File: build.gradle [hunk 2]
```diff
@@ -138,7 +138,7 @@ tasks.register('integrationUITest', Test) {
         showFailedStandardStreams true
         showFullStackTraces true
     }
-    jvmArgs "-Djava.awt.headless=true"
+    jvmArgs "-Djava.awt.headless=false"
```
[the remark this answers is not in the document]
Comment (martinszuc) [reply build.gradle:141]: @lgrossma To enable access to clipboard used in AboutPublicTest. I used it for verifying the content of terminal within intellliJ

File: src/it/java/org/jboss/tools/intellij/openshift/test/ui/AbstractBaseTest.java (hunk 1 above)
Comment (martinszuc) [reply src/it/java/org/jboss/tools/intellij/openshift/test/ui/AbstractBaseTest.java:44]: @lgrossma I intend the individual Public or CLuster tests to inherit from these abstract classes:

```
AbstractBaseTest
├── AbstractPublicTest
│   ├── TestClass1 (Public Tests)
│   ├── TestClass2 (Public Tests)
│   └── ...
└── AbstractClusterTest
    ├── TestClass1 (Cluster Tests)
    ├── TestClass2 (Cluster Tests)
    └── ...

```
With this setup we only need the **first Test** to get RemoteRobot instance and create a project in intellij. From that point we can test the extension without every individual test to close and create project and the remote robot instance is common across all of the tests. 

So, to answer the question. The **_hasConnectedToTestIDE_** flag ensures that the test-project is created only once no matter which test is ran first. But WaitUntilBigTasksFinish runs before every test, so any code needed to run before each test would fall here.

I am not sure about the name of the flag or the connect method. Maybe refactor to setup before tests or something along those lines? (it was already there) Thoughts or suggestions?

File: gradle.properties [hunk 3]
```diff
@@ -1,4 +1,4 @@
-ideaVersion=IU-2023.3
+ideaVersion=IU-2023.2
```
[the remark this answers is not in the document]
Comment (martinszuc) [reply gradle.properties:1]: @lgrossma You are right. I ll revert this, my oversight :/

**Pull request closed without merging**